src/pydvl/utils/types.py

```diff
@@ -58,18 +83,10 @@ def maybe_add_argument(fun: Callable, new_arg: str) -> Callable:
     Returns:
         A new function accepting one more keyword argument.
     """
-    params = inspect.signature(fun).parameters
-    if new_arg in params.keys():
+    if new_arg in unroll_partial_fn_args(fun):
```

Comment:
What is the logic behind this change?

Reply:
Once a partial function is passed to `maybe_add_argument`, the inner ``__call__`` logic is neglected for these types and thus also if it is contained in the arguments. Furthermore `partial` has to be used, because otherwise the wrapped function can't be pickled (happens for `seed` but not for `job_id`). So recursive `partial` support is needed.